swarm/network/stream/snapshot_sync_test.go

```diff
@@ -0,0 +1,568 @@
+func BenchmarkSyncing_1024(b *testing.B) { benchmarkSyncing(b, 1024) }
+
+func benchmarkSyncing(b *testing.B, chunkCount int) {
+	for i := 0; i < b.N; i++ {
+		result, err := testSyncing(b.T, chunkCount)
+		if err != nil {
+			b.Fatalf("setting up simulation failed", result)
+		}
+		if result.Error != nil {
+			b.Logf("simulation failed: %s", result.Error)
+		}
+	}
+}
+*/
+
+func testSyncing(t *testing.T, chunkCount int, nodeCount int) {
+	ids = make([]discover.NodeID, nodeCount)
+	err := runSyncTest(chunkCount, nodeCount)
+	if err != nil {
+		t.Fatal(err)
+	}
+}
+
+/*
+The test generates the given number of chunks,
+then uploads these to a random node.
+Afterwards for every chunk generated, the nearest node addresses
+are identified, syncing is started, and finally we verify
+that the nodes closer to the chunk addresses actually do have
+the chunks in their local stores.
+
+The test loads a snapshot file to construct the swarm network,
+assuming that the snapshot file identifies a healthy
+kademlia network. The snapshot should have 'streamer' in its service list.
+*/
+func runSyncTest(chunkCount int, nodeCount int) error {
+
+	conf = &synctestConfig{}
+	//mapping of nearest node addresses for chunk hashes
+	//nodesToChunksMap = make(map[discover.NodeID][]storage.Key)
+	conf.retrievalMap = make(map[string]map[string]time.Duration)
+	conf.idToAddrMap = make(map[discover.NodeID][]byte)
+	conf.addrToIdMap = make(map[string]discover.NodeID)
+	//First load the snapshot from the file
+	net, err := initNetWithSnapshot(nodeCount)
+	if err != nil {
+		return err
+	}
+	defer net.Shutdown()
+
+	//get the nodes of the network
+	nodes := net.GetNodes()
+	//select one index at random...
+	idx := rand.Intn(len(nodes))
+	//...and get the the node at that index
+	//this is the node selected for upload
+	node := nodes[idx]
+	//iterate over all nodes...
+	for c := 0; c < len(nodes); c++ {
+		//create an array of discovery nodeIDS
+		ids[c] = nodes[c].ID()
+		//and a correspondent array of overlay addresses,
+		//later used for chunk proximity calculation
+		a := network.ToOverlayAddr(ids[c].Bytes())
+		conf.addrs = append(conf.addrs, a)
+		//the proximity calculation is on overlay addr,
+		//the p2p/simulations check func triggers on discover.NodeID,
+		//so we need to know which overlay addr maps to which nodeID
+		conf.idToAddrMap[ids[c]] = a
+		conf.addrToIdMap[string(a)] = ids[c]
+	}
+
+	ppmap = network.NewPeerPot(testMinProxBinSize, ids, conf.addrs)
+	// channel to signal simulation initialisation with action call complete
+	// or node disconnections
+	//disconnectC := make(chan error)
+	//quitC := make(chan struct{})
+
+	//after the test, clean up local stores initialized with createLocalStoreForId
+	defer localStoreCleanup()
+
+	trigger := make(chan discover.NodeID)
+	//triggerCheck defines what will be checked during the test
+	triggerCheck := func(ctx context.Context, id discover.NodeID) (bool, error) {
```

Comment:
just call it `check` pls

Reply:
pls put `action` before `check` as it is called first and logically easier to follow